src/tests/unit/tests/electron/ipc/async-action.test.ts

```diff
@@ -17,4 +17,16 @@ describe('AsyncAction', () => {
             mock.verify(m => m(), Times.once());
         }
     });
+
+    it("calls listeners in order they're created", async () => {
+        const testSubject = new AsyncAction();
+        const listenerMocks = [];
+        for (let i = 0; i <= 3; i++) {
+            listenerMocks.push(Mock.ofInstance(async () => {}));
+            testSubject.addAsyncListener(listenerMocks[i].object);
+        }
+        await testSubject.invokeAsync();
+
+        //any way to verify the order in which the mocks were called?
```

Comment:
Does anyone know of a way to verify that functions were called in a particular order? I'd also like to add that to the android-browser-close-cleanup-initializer.test.ts

Reply:
Rather than using `Mock.ofInstance(async () => {})`, you can use stub listeners that update a list (`async () => { listenerCallHistory.push(i) }`) and then write an expectaion against the list at the end.

If you want to verify not just that they're called in order but that each one is `await`ed in sequence before starting the next, you might instead do ```listenerCallHistory.push(`${i} start`); await tick(); listenerCallHistory.push(`${i} end`);```

Reply:
thanks @dbjorge this worked great!